lib/active_model/one_time_password.rb

```diff
@@ -5,20 +5,23 @@ module OneTimePassword
     module ClassMethods
 
       def has_one_time_password(options = {})
-
-        cattr_accessor :otp_column_name
-        class_attribute :otp_digits
+        cattr_accessor :otp_column_name, :otp_counter_column_name
+        class_attribute :otp_digits, :otp_counter_based
 
         self.otp_column_name = (options[:column_name] || "otp_secret_key").to_s
         self.otp_digits = options[:length] || 6
 
+        self.otp_counter_based = (options[:counter_based] || false)
+        self.otp_counter_column_name = (options[:counter_column_name] || "otp_counter").to_s
+
         include InstanceMethodsOnActivation
 
         before_create { self.otp_regenerate_secret if !self.otp_column}
+        before_create { self.otp_regenerate_counter if self.otp_counter_based && !self.otp_counter}
```

Comment:
Line is too long. [99/80]<br>Redundant `self` detected.<br>Space missing inside }.